labs/lab-4-jquery/table.js

```diff
@@ -0,0 +1,32 @@
+class Table {
+  constructor(x, y, id){
+    this.id = id;
+    this.selector = (`table-${this.id}`)
+    this.status = "avaliable";
+    this.positionX = x;
+    this.positionY = y;
+    this.party = null;
+  }
+  reserve(party) {
+    this.status = "reserved";
+    this.party = party;
+    $(`"${this.selector}`).css("background-color", "#7e7272");
+  }
+  unreserve() {
+    this.status = "avaliable";
+    this.party = null;
+  }
+  render() {
+    var button = $(`<div class="button-container" id="div-${this.selector}"><button class="table" id="${this.selector}">${this.id}</button></div>`);
+    $("#restaurant-wrapper").append(button);
+  }
+  initiateForm(){
+    var button = $(`<button class="table" id="${this.selector}">${this.id}</button>`)
```

Comment:
I think you want this `button` to be the `button` from `render` -- you've made a *new* button, that's not anywhere on the page, and hooked that button up to display the form.

You can either hook up this event in `render`, or you can store your rendered button in `this.button` or `this.element` or something like that, and then add the click event handler to that.